src/test/java/io/vinyldns/java/VinylDNSClientTest.java

```diff
@@ -1179,6 +1182,234 @@ public void createBatchChangeSuccess() {
+    batchResponse.setStatus(BatchChangeStatus.Rejected);
+    batchResponse.setApprovalStatus(BatchChangeApprovalStatus.ManuallyRejected);
+    batchResponse.setOwnerGroupId("someOwnerGroupId");
+    batchResponse.setReviewerId("reviewerId");
+
+    String reviewComment = "some rejection comment";
+    batchResponse.setReviewComment(reviewComment);
+    batchResponse.setReviewTimestamp(new Date());
+
+    String request = client.gson.toJson(new BatchChangeReview(reviewComment));
+    String response = client.gson.toJson(batchResponse);
+
+    wireMockServer.stubFor(
+        post(urlEqualTo("/zones/batchrecordchanges/" + batchResponse.getId() + "/reject"))
+            .withRequestBody(new EqualToJsonPattern(request, true, true))
+            .willReturn(
+                aResponse()
+                    .withStatus(202)
+                    .withHeader("Content-Type", "application/json")
+                    .withBody(response)));
+
+    VinylDNSResponse<BatchResponse> vinylDNSBatchResponse =
+        client.rejectBatchChanges("1234", reviewComment);
+
+    assertEquals(vinylDNSBatchResponse.getStatusCode(), 202);
+  }
+
+  @Test
+  public void rejectBatchChangeFailure() {
+    wireMockServer.stubFor(
+        post(urlEqualTo("/zones/batchrecordchanges/noExisto/reject"))
+            .willReturn(aResponse().withStatus(404).withBody("Not found")));
+
+    VinylDNSResponse<BatchResponse> vinylDNSResponse = client.rejectBatchChanges("noExisto");
+
+    assertTrue(vinylDNSResponse instanceof ResponseMarker.Failure);
+    assertEquals(vinylDNSResponse.getStatusCode(), 404);
+    assertEquals(vinylDNSResponse.getMessageBody(), "Not found");
+    assertNull(vinylDNSResponse.getValue());
+  }
+
+  @Test
+  public void cancelBatchChangeSuccess() {
+    List<SingleChange> singleChangeList = new ArrayList<>();
+
+    AddSingleChange addSingleChange = new AddSingleChange();
+    addSingleChange.setChangeType(ChangeInputType.Add);
+    addSingleChange.setId("1234");
+    addSingleChange.setInputName("testString");
+    addSingleChange.setRecord(new AData("1.2.3.4"));
+    addSingleChange.setType(RecordType.A);
+    addSingleChange.setStatus(SingleChangeStatus.Rejected);
+    singleChangeList.add(addSingleChange);
+
+    BatchResponse batchResponse = new BatchResponse();
+    batchResponse.setId("1234");
+    batchResponse.setUserId("testuser");
+    batchResponse.setUserName("testuser");
+    batchResponse.setCreatedTimestamp(new Date());
+    batchResponse.setChanges(singleChangeList);
+    batchResponse.setStatus(BatchChangeStatus.Cancelled);
+    batchResponse.setApprovalStatus(BatchChangeApprovalStatus.Cancelled);
+    batchResponse.setOwnerGroupId("someOwnerGroupId");
+    batchResponse.setReviewerId("reviewerId");
+    batchResponse.setReviewComment("Looks good!");
+    batchResponse.setReviewTimestamp(new Date());
+
+    String response = client.gson.toJson(batchResponse);
+
+    wireMockServer.stubFor(
```

Comment:
hmm, I see a stub for a cancel but I don't see the call being made, amirite?

Reply:
Yes, that is correct.